pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -590,31 +621,60 @@ func (i *inode) RmDir(ctx context.Context, name string, child kernfs.Inode) erro
 	return res.Error()
 }
 
+func (i *inode) Rename(ctx context.Context, oldname, newname string, child, dstDir kernfs.Inode) error {
+	fusefs := i.fs
+	task, creds := kernel.TaskFromContext(ctx), auth.CredentialsFromContext(ctx)
+
+	dstDirInode, ok := dstDir.(*inode)
+	if !ok {
+		log.Warningf("fusefs.Inode.Rename: Bad inode returned", dstDir)
+		return linuxerr.EINVAL
+	}
+
+	in := linux.FUSERenameIn{
+		Newdir: primitive.Uint64(dstDirInode.nodeID),
+		Oldname: linux.CString(oldname),
+		Newname: linux.CString(newname),
+	}
+	req := fusefs.conn.NewRequest(creds, uint32(task.ThreadID()), i.nodeID, linux.FUSE_RENAME, &in)
+	res, err := i.fs.conn.Call(task, req)
+	if err != nil {
+		return err
+	}
+	return res.Error()
```

Comment:
Since fuse built on top of kernfs - kernfs is responsible for (Dentry) dentries caching.
[Here](https://github.com/google/gvisor/blob/6e9aa169aca28a1e6312a5bf3125455682aa3953/pkg/sentry/fsimpl/kernfs/filesystem.go#L738), kernfs invalidates a cache after fuse.Rename returns, also the same is relevant to RmDir and others as well.